wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -30,6 +37,380 @@ sealed abstract class Wallet extends LockedWallet with UnlockedWalletApi {
+            val fundingInputs =
+              spendingInfoDbs
+                .map { db =>
+                  val info =
+                    db.toUTXOSpendingInfo(account, keyManager, dlcDb.network)
+                  (info.outPoint, info.output)
+                }
+            DLCOffer(
+              dlcDb.contractInfo,
+              oracleInfo,
+              keyManager.getRootXPub, // fixme this will need to be changed according to KeyDerivation.md
+              amount.satoshis,
+              fundingInputs,
+              Bech32Address(dlcDb.changeSPK, dlcDb.network),
+              feeRate,
+              dlcDb.timeouts
+            )
+          }
+        case None =>
+          logger.debug("DLC Offer not found, creating a new one")
+          createNewDLCOffer(account = account,
+                            eventId = eventId,
+                            amount = amount,
+                            oracleInfo = oracleInfo,
+                            contractInfo = contractInfo,
+                            feeRate = feeRate,
+                            timeouts = timeouts)
+      }
+    } yield offer
+  }
+
+  private def createNewDLCOffer(
+      account: AccountDb,
+      eventId: Sha256DigestBE,
+      amount: Bitcoins,
+      oracleInfo: OracleInfo,
+      contractInfo: ContractInfo,
+      feeRate: SatoshisPerVirtualByte,
+      timeouts: DLCTimeouts): Future[DLCOffer] = {
+    val dlcDbFs = for {
+      txBuilder <- fundRawTransactionInternal(
+        destinations = getDummyDLCFundingOutputs(amount),
+        feeRate = feeRate,
+        fromAccount = account,
+        keyManagerOpt = Some(keyManager)
+      )
+    } yield {
+
+      val utxos = txBuilder.utxoMap
+        .filter(_._2.amount != Satoshis.one)
+        .map(utxo => (utxo._1, utxo._2.output))
+        .toVector
+
+      val changeSPK =
+        txBuilder.changeSPK.asInstanceOf[WitnessScriptPubKey]
+
+      val network = networkParameters.asInstanceOf[BitcoinNetwork]
+
+      logger.debug("DLC Offer data collected, creating database entry")
+      (dlcDAO
+         .create(
+           DLCDb(
+             eventId = eventId,
+             oracleInfo = oracleInfo,
+             contractInfo = contractInfo,
+             timeouts = timeouts,
+             isInitiator = true,
+             remoteExtPubKey = ExtPrivateKey
+               .freshRootKey(SegWitTestNet3Priv)
+               .extPublicKey,
+             input = amount,
+             remoteInput = Satoshis.zero,
+             fundingInputs = utxos.map(_._1),
+             remoteFundingInputs = Vector.empty,
+             feeRate = feeRate,
+             changeSPK = changeSPK,
+             remoteChangeSPK = P2WSHWitnessSPKV0(EmptyScriptPubKey),
+             cetSigs = CETSignatures(PartialSignature.dummyPartialSig,
+                                     PartialSignature.dummyPartialSig,
+                                     PartialSignature.dummyPartialSig),
```

Comment:
Any reason not to make this an option in the database rather than filling it with dummy sigs? Not sure which I prefer tbh (cc @Christewart)

Reply:
same with the rest of the filler in this case class. Specifically I wonder if maybe a better design is to split up the offer and accept portions and use foreign keys to link the split tables?

Reply:
Using options might be better but splitting offers and accepts doesn't really make much sense to me. Once the DLC is initiated they will need to hold the exact same data the only difference being `isInitiator`

Reply:
In my opinion this stuff should be modeled in the type system as different states in an ADT, with them all being options at the database level.